document/in_all_doc_id_set_iterator_bench_test.go

```diff
@@ -0,0 +1,214 @@
+package document
+
+import (
+	"testing"
+
+	rroaring "github.com/RoaringBitmap/roaring"
+	proaring "github.com/pilosa/pilosa/roaring"
+	"github.com/willf/bitset"
+)
+
+func BenchmarkInAllDocIDSetIteratorDense(b *testing.B) {
+	benchmarkInAllDocIDSetIterator(b, testDenseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorBitsetDense(b *testing.B) {
+	benchmarkDocIDSetIteratorBitset(b, testDenseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorRRoaringBitmapDense(b *testing.B) {
+	benchmarkDocIDSetIteratorRRoaringBitmap(b, testDenseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorPilosaRoaringBitmapDense(b *testing.B) {
+	benchmarkDocIDSetIteratorPilosaRoaringBitmap(b, testDenseBenchSetup)
+}
+
+func BenchmarkInAllDocIDSetIteratorSparse(b *testing.B) {
+	benchmarkInAllDocIDSetIterator(b, testSparseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorBitsetSparse(b *testing.B) {
+	benchmarkDocIDSetIteratorBitset(b, testSparseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorRRoaringBitmapSparse(b *testing.B) {
+	benchmarkDocIDSetIteratorRRoaringBitmap(b, testSparseBenchSetup)
+}
+
+func BenchmarkDocIDSetIteratorPilosaRoaringBitmapSparse(b *testing.B) {
+	benchmarkDocIDSetIteratorPilosaRoaringBitmap(b, testSparseBenchSetup)
+}
+
+func benchmarkInAllDocIDSetIterator(b *testing.B, ts testBenchSetup) {
+	bms := initBenchBitmaps(ts)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		iters := make([]DocIDSetIterator, 0, ts.numFields)
+		for _, bm := range bms {
+			iters = append(iters, newbitmapBasedDocIDIter(bm.Iterator()))
+		}
+		inAllIter := newInAllDocIDSetIterIterator(iters...)
+
+		count := 0
+		for inAllIter.Next() {
+			count++
```

Comment:
why is the count needed?

Reply:
Was used to verify correctness, can remove.